bin/reth/src/dirs.rs

```diff
@@ -184,3 +184,10 @@ impl<D> From<PlatformPath<D>> for PathBuf {
         value.0
     }
 }
+
+impl<D> PlatformPath<D> {
```

Comment:
@gakonst do you mean moving the join method here or appending the network directly to here https://github.com/paradigmxyz/reth/pull/1846/files#diff-7f870f125b66bc2e3781183e1af7ab859e1379a35f111120392cf45c621b0f97L21 ? 

If we would want to append the network directly to the `database_path` function we would somehow need to access the chain configuration

```rust
pub fn database_path() -> Option<PathBuf> {
    data_dir().map(|root| root.join("db"))
}
``` 

What do you think makes more sense?

Reply:
yep agreed, we'd want a way for the db path function to access `Chain`

Reply:
The methods `data_dir`, `database_path`, etc. are all used in `<T as XdgPath>::resolve()`, which is how `PlatformPath<T>` can implement `Default` in the first place. So these path methods can't really take any arguments.

We should probably do #2179 instead, because then we could then have a common `Default` impl for the root data directory, and append the chain to that directory. This would let child directories (`db`, `net`, `p2p`, ...) be per-chain, and could avoid the problem of `PlatformPath<T>::default()` not having access to the `Chain`.